include/boost/geometry/algorithms/detail/overlay/get_turns.hpp

```diff
@@ -482,11 +492,13 @@ class get_turns_generic
 
         typename IntersectionStrategy::envelope_strategy_type const
             envelope_strategy = intersection_strategy.get_envelope_strategy();
```

Comment:
is it needed for intersection strategy to contain both envelope and expand strategies? Why not passing them independently as in the strategies in `disjoint_segment_box`?